File: backend/src/main/java/com/example/Calculator.java
package com.example;

/**
 * A simple calculator class for testing purposes
 */
public class Calculator {

    /**
     * Adds two integers
     * @param a first number
     * @param b second number
     * @return sum of a and b
     */
    public int add(int a, int b) {
        return a + b;
    }

    /**
     * Subtracts b from a
     * @param a first number
     * @param b second number
     * @return difference of a and b
     */
    public int subtract(int a, int b) {
        return a - b;
    }

    /**
     * Multiplies two integers
     * @param a first number
     * @param b second number
     * @return product of a and b
     */
    public int multiply(int a, int b) {
        return a * b;
    }

    /**
     * Divides a by b
     * @param a dividend
     * @param b divisor
     * @return quotient of a and b
     * @throws ArithmeticException if b is zero
     */
    public double divide(int a, int b) {
        if (b == 0) {
            throw new ArithmeticException("Division by zero is not allowed");
        }
        return (double) a / b;
    }

    /**
     * Checks if a number is even
     * @param number the number to check
     * @return true if even, false otherwise
     */
    public boolean isEven(int number) {
        return number % 2 == 0;
    }
}

File: frontend/src/Calculator.ts
/**
 * A simple calculator class for TypeScript
 */
export class Calculator {
    /**
     * Adds two numbers
     */
    add(a: number, b: number): number {
        return a + b;
    }

    /**
     * Subtracts b from a
     */
    subtract(a: number, b: number): number {
        return a - b;
    }

    /**
     * Multiplies two numbers
     */
    multiply(a: number, b: number): number {
        return a * b;
    }

    /**
     * Divides a by b
     */
    divide(a: number, b: number): number {
        if (b === 0) {
            throw new Error('Division by zero is not allowed');
        }
        return a / b;
    }

    /**
     * Checks if a number is even
     */
    isEven(num: number): boolean {
        return num % 2 === 0;
    }

    /**
     * Calculates the factorial of a number
     */
    factorial(n: number): number {
        if (n < 0) {
            throw new Error('Factorial is not defined for negative numbers');
        }
        if (n === 0 || n === 1) {
            return 1;
        }
        return n * this.factorial(n - 1);
    }
}
